storage_drivers/azure/api/azure.go

```diff
@@ -129,17 +131,49 @@ type Client struct {
 }
 
 // NewDriver is a factory method for creating a new SDK interface.
-func NewDriver(config ClientConfig) (Azure, error) {
+func NewDriver(ctx context.Context, config ClientConfig) (Azure, error) {
 	var err error
+	var credential azcore.TokenCredential
 
 	// Ensure we got a location
 	if config.Location == "" {
 		return nil, errors.New("location must be specified in the config")
 	}
 
-	credential, err := azidentity.NewClientSecretCredential(config.TenantID, config.ClientID, config.ClientSecret, nil)
-	if err != nil {
-		return nil, err
+	if credFilePath := os.Getenv("AZURE_CREDENTIAL_FILE"); credFilePath != "" {
+		Logc(ctx).WithField("credFilePath", credFilePath).Info("Using Azure credentials from file.")
+		var azureConfig *azure.Config
+		credFile, err := os.Open(credFilePath)
+		if err != nil {
+			return nil, errors.New("error opening azure config file: " + err.Error())
+		}
+		defer credFile.Close()
+		if azureConfig, err = azure.ParseConfig(credFile); err != nil {
+			return nil, errors.New("error parsing azure config file: " + err.Error())
+		}
+		// use managed identity credential
+		opts := azidentity.ManagedIdentityCredentialOptions{ID: azidentity.ClientID(azureConfig.UserAssignedIdentityID)}
```

Comment:
the azure.json could also contain service principal in `clientID` and `clientSecret`, so the logic could be: if backend secret is not defined, then parse azure.jjson file to get credentials.

Reply:
Yes, however, I didn't find any avalaible method in cloud-provider-azure that can parse the credential file and return corresponding credential. I will see if I can make it in cloud-provider-azure SDK.

Reply:
the cloud provider config parsing using old sdk is here: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/470a363a5dbb295bd1123d9cf8cb0e34ac87b032/pkg/provider/config/azure_auth.go#L95, the parsing logic should be the same

Reply:
maybe we could leverage the sdk 2 parsing code directly: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/470a363a5dbb295bd1123d9cf8cb0e34ac87b032/pkg/azclient/auth.go
cc @MartinForReal

Reply:
Yes, I've made some changes to the code. However, there seems like some problems when using go-armbalancer, without it everything is ok. I'm investigating in it.